### PR TITLE
[RISC-V] Fix some minor errors

### PR DESCRIPTION
**Pull request by clamp03**
Fix some minor errors in RISC-V codes.

Part of https://github.com/dotnet/runtime/issues/84834
cc @dotnet/samsung

### CODE REVIEW

**Comment by risc-vv** [comment]
<details><summary>8d8d85f0 is being scheduled for building and testing</summary>

GIT: `8d8d85f034844989842771ae69b9a3a59c89927d`
REPO: [clamp03/runtime](https://github.com/clamp03/runtime)
BRANCH: legacyexception</details>

**Comment by clamp03** [comment]
@janvorli In test45929, it fails 1~4 out of 500 times in early August. However, recently it fails more. (*. 1 ~2 out of 5 times on my VF2 machine.)

Error message (release build) is 
```
Process terminated. UnhandledException
   at System.Environment.FailFast(System.Runtime.CompilerServices.StackCrawlMarkHandle, System.String, System.Runtime.CompilerServices.ObjectHandleOnStack, System.String)
   at System.Environment.FailFast(System.Threading.StackCrawlMark ByRef, System.String, System.Exception, System.String)
   at System.Environment.FailFast(System.String)
   at System.Runtime.EH.FallbackFailFast(System.Runtime.RhFailFastReason, System.Object)
   at System.Runtime.EH.UnhandledExceptionFailFastViaClasslib(System.Runtime.RhFailFastReason, System.Object, IntPtr, ExInfo ByRef)
   at System.Runtime.EH.DispatchEx(System.Runtime.StackFrameIterator ByRef, ExInfo ByRef)
   at System.Runtime.EH.RhThrowEx(System.Object, ExInfo ByRef)
   at test45929.Program+Test+TestCore.ExceptionDispatchInfoCaptureThrow()
./Regressions/coreclr/GitHub_45929/test45929/test45929.sh: line 441: 467131 Aborted                 (core dumped) $LAUNCHER $ExePath "${CLRTestExecutionArguments[@]}"                                                                        
Expected: 100
Actual: 134
END EXECUTION - FAILED
```

Backtrace in coredump is 
```
#0  0x0000003faa8d961a in __pthread_kill_implementation (threadid=<optimized out>, signo=signo@entry=6, no_tid=no_tid@entry=0) at pthread_kill.c:44
#1  0x0000003faa8d9650 in __pthread_kill_internal (threadid=<optimized out>, signo=6) at pthread_kill.c:78
#2  0x0000003faa8a3264 in __GI_raise (sig=sig@entry=6) at ../sysdeps/posix/raise.c:26
#3  0x0000003faa89420c in __GI_abort () at abort.c:79
#4  0x0000003faa60d5f8 in PROCAbort (signal=<optimized out>, siginfo=<optimized out>) at /home/clamp/Work/dotnet/srpol/567/runtime.upstream/src/coreclr/pal/src/thread/process.cpp:2562
#5  0x0000003faa60d520 in PROCEndProcess (hProcess=<optimized out>, uExitCode=2148734499, bTerminateUnconditionally=1) at /home/clamp/Work/dotnet/srpol/567/runtime.upstream/src/coreclr/pal/src/thread/process.cpp:1354
#6  TerminateProcess (hProcess=<optimized out>, uExitCode=2148734499) at /home/clamp/Work/dotnet/srpol/567/runtime.upstream/src/coreclr/pal/src/thread/process.cpp:1251
#7  0x0000003faa379f06 in EEPolicy::HandleFatalError (exitCode=2148734499, address=271310995612, pszMessage=0x37d5c0457c u"UnhandledException", pExceptionInfo=0x37b17c9580, errorSource=0x0, argExceptionString=0x0)
    at /home/clamp/Work/dotnet/srpol/567/runtime.upstream/src/coreclr/vm/eepolicy.cpp:828
#8  0x0000003faa5a8bf2 in Environment_FailFast (mark=..., message=0x37d5c0457c u"UnhandledException", exception=..., errorSource=0x0) at /home/clamp/Work/dotnet/srpol/567/runtime.upstream/src/coreclr/classlibnative/bcltype/system.cpp:198
#9  0x0000003f2b3da364 in ?? ()
```

I checked that new exception handling is ported from NativeAOT. However, we didn't implement NativeAOT for RISC-V yet. (*. @am11 is working on it. https://github.com/dotnet/runtime/issues/106223 Thank you.)
In my guess, new exception handling is not fully working on RISC-V. Am I right? Then to support new exception handling, should we implement NativeAOT fully? Could you let me know necessary code region to support only new exception handling?
Thank you.

**Comment by janvorli** [comment]
> In my guess, new exception handling is not fully working on RISC-V.

While it was ported from NativeAOT, it is not dependent on other NativeAOT code, so whether NativeAOT is implemented or not doesn't matter for this case. 
It is possible that there are some issues with RISC-V. Since the failure is intermittent, it seems it is not a general problem with the new EH, as it would then be failing in many more EH tests. This test that's failing is quite special, as it is very heavily multi-threaded, basically throwing and catching exceptions on as many threads in parallel as possible. So there could be some threading issue specific to RISC-V somewhere in the runtime that just gets exposed by this test.
I would not recommend to disable the new EH, as we are planning to remove the old EH completely, maybe in the .NET 10 timeframe. I'd rather disable this specific test for RISC-V for now if you want to keep the test runs clean and figure out what's the culprit here. The error means that the exception handling didn't find exception handler for the exception, but in a way that should not happen.

**Comment by clamp03** [comment]
Thank you for your comment. I will investigate more and try the others. Thank you.

**Comment by risc-vv** [comment]
<details><summary>5fcc9ff7 is being scheduled for building and testing</summary>

GIT: `5fcc9ff7b15b267f49119b3c48c319803d85c038`
REPO: [clamp03/runtime](https://github.com/clamp03/runtime)
BRANCH: legacyexception</details>

**Comment by risc-vv** [comment]
<details><summary>deeffb6e is being scheduled for building and testing</summary>

GIT: `deeffb6ee296a2c4de4fd1f060f4d36d1a588891`
REPO: [clamp03/runtime](https://github.com/clamp03/runtime)
BRANCH: legacyexception</details>

**Comment by risc-vv** [comment]
<details><summary>fecdcb9c is being scheduled for building and testing</summary>

GIT: `fecdcb9c5aa27d66235b4104d8e6492f10ebdb46`
REPO: [clamp03/runtime](https://github.com/clamp03/runtime)
BRANCH: legacyexception</details>

**Comment by risc-vv** [comment]
<details><summary>43c270c2 is being scheduled for building and testing</summary>

GIT: `43c270c2dfb11c3002e4b0981cc5eef63c53104d`
REPO: [clamp03/runtime](https://github.com/clamp03/runtime)
BRANCH: legacyexception</details>

File: src/coreclr/unwinder/riscv64/unwinder.cpp [hunk 1]
```diff
@@ -647,6 +647,7 @@ Return Value:
     ULONG UnwindIndex;
     ULONG UnwindWords;
 
+    assert(!"---------------RISCV64 ShouldNotReachHere");
```
Comment (jkotas) [inline src/coreclr/unwinder/riscv64/unwinder.cpp:650]: Delete this method instead?

**Comment by jkotas** [comment]
Could you please resolve the merge conflict?

**Review by viewizard — APPROVED** [review]

**Comment by risc-vv** [comment]
<details><summary>c4e6b328 is being scheduled for building and testing</summary>

GIT: `c4e6b328ba9c6fec2310f20129a16a14379b5d9b`
REPO: [dotnet/runtime](https://github.com/dotnet/runtime)
BRANCH: main</details>

**Comment by risc-vv** [comment]
<details><summary>7090560b is being scheduled for building and testing</summary>

GIT: `7090560b0924ac813dd1a29b253ea5d77ad7e5b1`
REPO: [dotnet/runtime](https://github.com/dotnet/runtime)
BRANCH: main</details>

File: src/coreclr/unwinder/riscv64/unwinder.cpp (hunk 1 above)
Comment (clamp03) [reply src/coreclr/unwinder/riscv64/unwinder.cpp]: Oops. I want to put this into `RtlpUnwindCustom` not `RtlpUnwindFunctionFull`.
I will update `unwinder.cpp` and then I will request your review again.
Thank you.

**Comment by risc-vv** [comment]
<details><summary>83ba9788 is being scheduled for building and testing</summary>

GIT: `83ba978844c04683d5c064866373262384a192f7`
REPO: [dotnet/runtime](https://github.com/dotnet/runtime)
BRANCH: main</details>

**Review by janvorli — APPROVED** [review]
LGTM, thank you!

**Comment by am11** [comment]
innerloop leg timedout https://github.com/dotnet/dnceng/issues/3879 (lack of RAM for -allConfiguration leg which builds assemblies for all platform combinations, which I think should be split into multiple invocation of `dotnet` so csc.dll keeps releasing memory)

**Pull request merged**